File: codegen/codegen-erlang/src/test/resources/ir/s3_endpoint_helpers.expected.erl
key_path(<<>>) -> <<>>;
key_path(Key) -> <<"/", Key/binary>>.

virtual_host(Config, Bucket, RegionHost) ->
    case maps:get(s3_use_accelerate, Config, false) of
        true -> <<Bucket/binary, ".s3-accelerate.amazonaws.com">>;
        false ->
            Suffix = s3_host_suffix(Config),
            <<Bucket/binary, Suffix/binary, RegionHost/binary>>
    end.

s3_host_suffix(Config) ->
    case maps:get(s3_use_dualstack, Config, false) of
        true -> <<".s3.dualstack.">>;
        false -> <<".s3.">>
    end.

File: codegen/codegen-elixir/src/test/resources/ir/s3_endpoint_helpers.expected.ex
defp key_path(""), do: ""
defp key_path(key), do: "/#{key}"

defp virtual_host(config, bucket, region_host) do
  if Map.get(config, :s3_use_accelerate, false) do
    "#{bucket}.s3-accelerate.amazonaws.com"
  else
    suffix = s3_host_suffix(config)
    "#{bucket}#{suffix}#{region_host}"
  end
end

defp s3_host_suffix(config) do
  if Map.get(config, :s3_use_dualstack, false), do: ".s3.dualstack.", else: ".s3."
end
